server/internal/infrastructure/tx-builder/covenantless/builder.go

```diff
@@ -329,28 +328,59 @@ func (b *txBuilder) VerifyForfeitTxs(
 			continue
 		}
 
-		connectorInput := tx.UnsignedTx.TxIn[0]
-		vtxoInput := tx.UnsignedTx.TxIn[1]
+		var vtxoInput, connectorInput *wire.TxIn
+		var vtxoTapscript *psbt.TaprootTapLeafScript
+		var connectorOutput *wire.TxOut
+		var vtxoFirst bool
+
+		// search for the connector output and the vtxo input in the tx
+		for i, input := range tx.UnsignedTx.TxIn {
+			for _, connector := range connectorsLeaves {
+				if connector.Txid == input.PreviousOutPoint.Hash.String() {
+					connectorTx, err := psbt.NewFromRawBytes(strings.NewReader(connector.Tx), true)
+					if err != nil {
+						return nil, err
+					}
+
+					if len(connectorTx.UnsignedTx.TxOut) <= int(input.PreviousOutPoint.Index) {
+						continue
```

Comment:
shall we return an error?
```suggestion
						return nil, fmt.Errorf("connector vout %d out of range [0, %d]", input.PreviousOutPoint.Index), len(connectorTx.UnsignedTx.TxOut) -1)
```